app/policies/project_policy.rb

```diff
@@ -0,0 +1,31 @@
+class ProjectPolicy < ApplicationPolicy
+  authorize :user, allow_nil: true
+
+  def index?
+    true
+  end
+
+  def show?
+    true
+  end
+
+  def new?
+    create?
+  end
+
+  def create?
+    user.present?
+  end
+
+  def edit?
+    update?
+  end
+
+  def update?
+    record.project_memberships.find_by(user: user).role = 'member' && user.present?
```

Comment:
Check user presence first or it is redundant condition. Owner of the project should can edit project too